internal/inspect/detectors/quarto.go

```diff
@@ -257,42 +257,119 @@ func (d *QuartoDetector) includeStaticConfig(base util.AbsolutePath, cfg *config
 		return
 	}
 
-	staticCfg := config.New()
-	staticCfg.Type = config.ContentTypeHTML
-	staticCfg.Title = cfg.Title
+	d.log.Debug("Generating Quarto static configuration", "entrypoint", cfg.Entrypoint)
 
 	// With output-dir present, we know that any static asset will be in that directory.
 	// If it is a website project, quarto inspect sets _site as the default output directory.
-	outputDir := inspectOutput.OutputDir()
-	if outputDir != "" {
-		staticCfg.Entrypoint = outputDir
-		staticCfg.Files = []string{fmt.Sprint("/", outputDir)}
-		cfg.Alternatives = append(cfg.Alternatives, *staticCfg)
+	if inspectOutput.OutputDir() != "" {
+		staticCfg := d.staticConfigFromOutputDir(base, cfg, inspectOutput)
+		if staticCfg != nil {
+			cfg.Alternatives = append(cfg.Alternatives, *staticCfg)
+		}
 		return
 	}
 
+	d.log.Debug("Quarto project does not specify an output-dir")
+	d.log.Debug("Attempting to generate Quarto document static configuration from HTML files in project")
+
 	// Last option to generate a static configuration,
 	// use inspect output files.input to generate a list with the .html ext version of the files.
 	// (If there is a render list, files.input is already filtered down to the applicable files.)
+	staticCfg := d.staticConfigFromFilesLookup(base, cfg, inspectOutput)
+	if staticCfg != nil {
+		cfg.Alternatives = append(cfg.Alternatives, *staticCfg)
+	}
+}
+
+// Generate a static configuration based on the output-dir specified in the project,
+// with discoverability and generation of the entrypoint HTML file related to the output-dir.
+func (d *QuartoDetector) staticConfigFromOutputDir(base util.AbsolutePath, cfg *config.Config, inspectOutput *quartoInspectOutput) *config.Config {
```

Comment:
Changes here are mostly related to choosing an appropriate entrypoint when there is an `output-dir`.